src/transaction/log_replication.cpp

```diff
@@ -225,4 +276,41 @@ namespace cublog
 	m_parallel_replication_redo->wait_for_idle ();
       }
   }
+
+  /* log_rpl_calculate_replication_delay - calculate delay based on a given start time value
+   *        and the current time and log to the perfmon infrastructure; all calculations are
+   *        done in milliseconds as that is the relevant scale needed
+   *
+   *  a_start_time_source: the name of the source where this
+   */
+  int
+  log_rpl_calculate_replication_delay (THREAD_ENTRY *thread_p, time_t a_start_time_msec,
+				       const char *a_start_time_source)
+  {
+    // skip calculation if bogus input (sometimes, it is -1);
+    // TODO: fix bogus input at the source if at all possible
+    if (a_start_time_msec > 0)

```

Comment:
Does it happen to get bogus start time?

Reply:
Sometimes yes, that is one of the two reasons I added the `m_source`.
TODO: add logging snapshot

Reply:
If this is an outlier event, I think we can just ignore it. I don't see anything that can be done with this error.